src/dosbox.cpp

```diff
@@ -573,46 +569,52 @@ void DOSBOX_Init(void) {
+	const char *ios[] = {"220", "240", "260", "280", "2a0", "2c0", "2e0", "300", 0};
+	Phex = secprop->Add_hex("sbbase", Property::Changeable::WhenIdle, 0x220);
 	Phex->Set_values(ios);
-	Phex->Set_help("The IO address of the soundblaster.");
+	Phex->Set_help("The IO address of the Sound Blaster.");
 
-	Pint = secprop->Add_int("irq",Property::Changeable::WhenIdle,7);
+	const char *irqssb[] = {"7", "5", "3", "9", "10", "11", "12", 0};
+	Pint = secprop->Add_int("irq", Property::Changeable::WhenIdle, 7);
 	Pint->Set_values(irqssb);
-	Pint->Set_help("The IRQ number of the soundblaster.");
+	Pint->Set_help("The IRQ number of the Sound Blaster.");
 
-	Pint = secprop->Add_int("dma",Property::Changeable::WhenIdle,1);
+	const char *dmassb[] = {"1", "5", "0", "3", "6", "7", 0};
+	Pint = secprop->Add_int("dma", Property::Changeable::WhenIdle, 1);
 	Pint->Set_values(dmassb);
-	Pint->Set_help("The DMA number of the soundblaster.");
+	Pint->Set_help("The DMA number of the Sound Blaster.");
 
-	Pint = secprop->Add_int("hdma",Property::Changeable::WhenIdle,5);
+	Pint = secprop->Add_int("hdma", Property::Changeable::WhenIdle, 5);
 	Pint->Set_values(dmassb);
-	Pint->Set_help("The High DMA number of the soundblaster.");
+	Pint->Set_help("The High DMA number of the Sound Blaster.");
 
-	Pbool = secprop->Add_bool("sbmixer",Property::Changeable::WhenIdle,true);
-	Pbool->Set_help("Allow the soundblaster mixer to modify the DOSBox mixer.");
+	Pbool = secprop->Add_bool("sbmixer", Property::Changeable::WhenIdle, true);
+	Pbool->Set_help("Allow the Sound Blaster mixer to modify the DOSBox mixer.");
 
-	const char* oplmodes[]={ "auto", "cms", "opl2", "dualopl2", "opl3", "opl3gold", "none", 0};
-	Pstring = secprop->Add_string("oplmode",Property::Changeable::WhenIdle,"auto");
+	const char* oplmodes[] = {"auto", "cms", "opl2", "dualopl2", "opl3", "opl3gold", "none", 0};
+	Pstring = secprop->Add_string("oplmode", Property::Changeable::WhenIdle, "auto");
 	Pstring->Set_values(oplmodes);
-	Pstring->Set_help("Type of OPL emulation. On 'auto' the mode is determined by sblaster type. All OPL modes are Adlib-compatible, except for 'cms'.");
+	Pstring->Set_help("Type of OPL emulation. On 'auto' the mode is determined by 'sbtype'.\n"
+	                  "All OPL modes are AdLib-compatible, except for 'cms'.");
 
-	const char* oplemus[]={ "default", "compat", "fast", "mame", 0};
-	Pstring = secprop->Add_string("oplemu",Property::Changeable::WhenIdle,"default");
+	const char* oplemus[] = {"default", "compat", "fast", "mame", "nuked", 0};
+	Pstring = secprop->Add_string("oplemu", Property::Changeable::WhenIdle, "default");
 	Pstring->Set_values(oplemus);
-	Pstring->Set_help("Provider for the OPL emulation. compat might provide better quality (see oplrate as well).");
+	Pstring->Set_help("Provider for the OPL emulation. 'compat' provides better quality,\n"
+	                  "'nuked' is the most accurate (see oplrate as well).");
```

Comment:
'nuked' is the most accurate and CPU intensive
(on my Rpi3 I have to disabled 'nuked' because it's too CPU intensive compared to the others.  Looking forward to getting a Pi4 when I have time, and suspect it will have enough cycles to support it)

Reply:
> 1. Looking at `make_opl_handler`, `default` isn't explicitly handled and the the function falls through to returning the `fast` OPL implementation.  We should mention that `fast` is the default (or perhaps eliminate `default` all-together?).

I would prefer to keep the "default" as a possible value without explicitly mentioning what does it point to in .conf file, as this allows us to change the default in the future without making .conf files generated for users incorrect.

> 2. If nuked is now cycle-accurate based on die shots (basically as perfect as it can get, besides finding coding bugs) then does that mean `nuked` by definition must also be the highest quality implementation of the bunch - if we're talking about emulation correctness? Seems like `compat` no longer deserves the _quality-crown_.  If `compat` goes above-and-beyond whatever flaws exist in the hardware, then maybe we can mention it's "enhanced" or "improved" versus the actual hardware?

Good suggestion… perhaps someone of Vogons can help us with assesing relative quality benefits of various implementations?

>  3. 'nuked' is the most accurate _and CPU intensive_ (...) Suggest adding the latter point. On my Rpi3 I have to disabled 'nuked' because it's too CPU intensive compared to the others.  Looking forward to getting a Pi4 when I have time, and suspect it will have enough cycles to support it.

Done :)

Reply:
Agree!